crates/interpreter/src/interpreter/stack.rs

```diff
@@ -222,8 +222,14 @@ impl Stack {
     }
 
     /// Duplicates the `N`th value from the top of the stack.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `n` is 0.
     #[inline]
+    #[cfg_attr(debug_assertions, track_caller)]
     pub fn dup(&mut self, n: usize) -> Result<(), InstructionResult> {
+        assume!(n > 0, "attempted to dup 0");
```

Comment:
What does `assume` do here, if we are checking `n >0`, as N is usize, it always will be more than 0?

Reply:
n == 0 doesn't make sense, there is no dup0 or swap0